src/DependencyInjection/Compiler/AdminMakerCompilerPass.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * This class injects available admin managers to the AdminMaker.
+ *
+ * @author Gaurav Singh Faudjdar <faujdar@gmail.com>
+ */
+final class AdminMakerCompilerPass implements CompilerPassInterface
+{
+    const MANAGERS = [
+        'sonata.admin.manager.orm',
+        'sonata.admin.manager.doctrine_mongodb',
+        'sonata.admin.manager.doctrine_phpcr',
+    ];
+
+    public function process(ContainerBuilder $container)
+    {
+        $availableManagers = [];
+        foreach (self::MANAGERS as $manager) {
+            if ($container->hasDefinition($manager)) {
+                $availableManagers[$manager] = $container->getDefinition($manager);
+            }
+        }
+
+        $definition = $container->getDefinition('sonata.admin.maker');
+        $definition->setArgument(0, $container->getParameter('kernel.project_dir'));
+        $definition->setArgument(1, $availableManagers);
```

Comment:
Here is the problem:

https://github.com/symfony/dependency-injection/blob/54243abc4e1a1a15e274e391bd6f7090b44711f1/Definition.php#L279

`setArgument` was added in sf3.3

Reply:
cc @OskarStark

Reply:
should i use https://github.com/symfony/dependency-injection/blob/1534e480b2d18bf1b0acb05271bab75118d022c2/Definition.php#L199